Sources/SwiftParser/Expressions.swift

```diff
@@ -1237,8 +1241,14 @@ extension Parser {
 
 extension Parser {
   /// Parse an identifier as an expression.
-  mutating func parseIdentifierExpression() -> RawExprSyntax {
-    let declName = self.parseDeclReferenceExpr(.compoundNames)
+  mutating func parseIdentifierExpression(flavor: ExprFlavor) -> RawExprSyntax {
+    var options: DeclNameOptions = .compoundNames
+    switch flavor {
+    case .basic, .poundIfDirective, .stmtCondition: break
+    case .attributeArgument: options.insert(.keywords)
```

Comment:
This does change the behavior a little bit more and allows eg. `names: named(class)`. I’m not sure if we want this or if we want to require backticks around `class`.

Reply:
Ah, so I actually wanted `keywordsUsingSpecialNames` here.

Reply:
That too but AFAICT `keywordsUsingSpecialNames` will still parse of `named(class)` with `class` as an identifier. Currently `class` needs to be backticked. We might want this behavior or not. I’m not sure.

Reply:
No, `keywordsUsingSpecialNames` is only for `subscript` and `deinit`. It was made just for this carve-out

Reply:
Oh, looks like I was wrong because the case in `parsePriamryExpression` doesn’t have a case for `class` and thus doesn’t enter `parseIdentifierExpression`. Sorry for the confusion.